### PR TITLE
Fix/4379 member acount balance

### LINKED ISSUE #4379: Show balance on `Accounts` tab
Show account balance on user profile sidebar (`Accounts` tab). The sidebar opens when user clicks on a member and helps to assess activity and reputation for example.

### PR DESCRIPTION
**Pull request by chrlschwb**
fix #4379

### CODE REVIEW

**Review by traumschule — CHANGES_REQUESTED** [review]
@chrlschwb 
 | [warn] src/common/components/BlockTime/BlockHistoryLine.tsx
 | [warn] src/memberships/components/MemberProfile/MemberAccounts.tsx
 | [warn] Code style issues found in 2 files. Forgot to run Prettier?
 | `yarn lint` failed with exit code 1

@thesan wrt storybook:
> Error: No existing credentials found. Please run `vercel login` or pass "--token"

File: packages/ui/src/memberships/components/MemberProfile/MemberAccounts.tsx [hunk 1]
```diff
@@ -6,35 +6,70 @@ import { AccountRow } from '../../../common/components/Modal'
...
-    )}
-  </AccountsDisplay>
-)
+  return (
+    < AccountsDisplay gap={16} >
+      <SidePaneLabel text="Root account" />
+      {!!member.rootAccount && (
+        <AccountMemberRow>
+          <UnknownAccountInfo address={member.rootAccount} placeholderName="Root Account" />
+          <TokenValue value={rootBalance?.total} isLoading={!isDefined(rootBalance?.total)} />
```
Comment (traumschule) [inline packages/ui/src/memberships/components/MemberProfile/MemberAccounts.tsx:23]: What could be the reason that it keeps blinking (never is defined)?

blinking

**Comment by HeinrichOlfert** [comment]
I have error when run `yarn lint:fix`

**Comment by chrlschwb** [comment]
image

I have error when run `yarn lint:fix`

**Comment by thesan** [comment]
@chrlschwb please get reviews from BWG workers ideally 2 of them. It would really help the Pioneer review flow.

File: packages/ui/src/common/modals/OnBoardingModal/OnBoardingPlugin.tsx [hunk 2]
```diff
@@ -38,12 +38,14 @@ export const OnBoardingPlugin = () => {
           <List>
             <ListItem>
               {getAllWallets().map((wallet) => (
-                <ConnectWalletItem
-                  key={wallet.extensionName}
-                  wallet={wallet}
-                  onClick={() => setSelectedWallet(wallet)}
-                  selected={selectedWallet?.extensionName === wallet.extensionName}
-                />
+                <>
```
Comment (kdembler) [inline packages/ui/src/common/modals/OnBoardingModal/OnBoardingPlugin.tsx:41]: This fragment doesn't seem needed?

File: packages/ui/src/memberships/components/MemberProfile/MemberAccounts.tsx [hunk 3]
```diff
@@ -1,40 +1,68 @@
...
+      )}
+    </AccountsDisplay>
+  )
+}
 
 const AccountsDisplay = styled(RowGapBlock)`
   padding: 24px;
 `
+
+export const AccountMemberRow = styled.div`
```
Comment (kdembler) [inline packages/ui/src/memberships/components/MemberProfile/MemberAccounts.tsx:52]: Why do we need to duplicate this style? We can use `AccountRow` as was imported before, and overwrite necessary CSS with `styled(AccountRow)`

File: packages/ui/src/memberships/components/MemberProfile/MemberBounds.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+export const MemberBounds = ({ account }: Props) => {
+  const balance = useBalance(account)
+  return (
+    <div>
+      {!!account && (
+        <AccountMemberRow>
+          <UnknownAccountInfo address={account} placeholderName="Root Account" />
+          <TokenValue value={balance?.total} isLoading={!isDefined(balance?.total)} />
+        </AccountMemberRow>
```
Comment (kdembler) [inline packages/ui/src/memberships/components/MemberProfile/MemberBounds.tsx:23]: This logic id duplicated 3 times: 1 one for root, 1 for controller and once in this component. I think we should adjust this component to be more generic and also use it for root and controller accounts

**Review by kdembler — CHANGES_REQUESTED** [review]
Left few comments. Also on visuals, I think the amount should be right-aligned, currently different values will be misaligned:
CleanShot 2023-11-13 at 10 28 08@2x

**Comment by kdembler** [comment]
Hi @chrlschwb my last comment wasn't resolved. What I was suggesting was to make `MemberBounds` into something like this:
```tsx
export const MemberAccount = ({ account, name }: Props) => {
  const balance = useBalance(account)
  return (
    <div>
      {!!account && (
        <AccountMemberRow>
          <UnknownAccountInfo address={account} placeholderName={name} />
          <TokenValue value={balance?.total} isLoading={!isDefined(balance?.total)} />
        </AccountMemberRow>
      )}
    </div>
  )
}
```
Then you can use it for all accounts, like root:
```tsx
      {!!member.rootAccount && (
        <MemberAccount account={member.rootAccount} name="Root account" />
      )}
```
This way you can also move style next to this component, removing circular dependency between two files

**Comment by chrlschwb** [comment]
> Hi @chrlschwb my last comment wasn't resolved. What I was suggesting was to make `MemberBounds` into something like this:
> 
> ```tsx
> export const MemberAccount = ({ account, name }: Props) => {
>   const balance = useBalance(account)
>   return (
>     <div>
>       {!!account && (
>         <AccountMemberRow>
>           <UnknownAccountInfo address={account} placeholderName={name} />
>           <TokenValue value={balance?.total} isLoading={!isDefined(balance?.total)} />
>         </AccountMemberRow>
>       )}
>     </div>
>   )
> }
> ```
> 
> Then you can use it for all accounts, like root:
> 
> ```tsx
>       {!!member.rootAccount && (
>         <MemberAccount account={member.rootAccount} name="Root account" />
>       )}
> ```
> 
> This way you can also move style next to this component, removing circular dependency between two files
year, I understand.

**Comment by chrlschwb** [comment]
year I understand.

**Review by kdembler — APPROVED** [review]
Could you just update the `MemberBounds` component and file name now that it's generic? Otherwise, LGTM

**Comment by chrlschwb** [comment]
what do you think the name?

**Comment by chrlschwb** [comment]
> Could you just update the `MemberBounds` component and file name now that it's generic? Otherwise, LGTM ?

what?

**Comment by kdembler** [comment]
I suggested `MemberAccount` earlier

**Review by thesan — APPROVED** [review]
Nice work, it looks great :+1:

**Comment by ivanturlakov** [comment]
Tested on https://pioneer-2.vercel.app/#/profile/memberships
Mainnet

✅ Balance added
Screenshot 2023-11-23 at 19 35 36

**Pull request merged**